superset/db_engine_specs.py

```diff
@@ -721,6 +721,70 @@ def get_table_names(cls, inspector, schema):
         return sorted(inspector.get_table_names())
 
 
+class DrillEngineSpec(BaseEngineSpec):
+    """Engine spec for Apache Drill"""
+    engine = 'drill'
```

Comment:
If you get around to trying out #7453 , try adding the following here:
```python
try_remove_schema_from_table_name = False
```
This will disable all attempts to remove schema from table names. While the risk of this happening are small (the table would have to be prefixed with the schema name followed by a period), it's better to disable schema name removal alltogether for engines that never add the schema name to table names.

Reply:
@villebro 
I'm going to wait until #7453 is committed.  My experiments with your branch basically solved all the issues with the Drill integration.  Once that's done, this PR will just be some mods to the `db_engine_specs.py` and hopefully will be easy to commit as well.

Reply:
I think I'm going to leave this out for the moment as Drill allows some schema variants that it might break.